tmt/base.py

```diff
@@ -516,11 +532,10 @@ def export(self, *, format_: ExportFormat = ExportFormat.YAML,
 
             # TODO: this belongs to Test.export, and it will be moved when the time
             # of export() cleanup comes.
-            elif key == 'require' and value:
+            elif key in ('require', 'recommend') and value:
                 data[key] = [
-                    require.to_minimal_dict() if isinstance(
-                        require, RequireFmfId) else require for require in cast(
-                        List[Require], value)]
+                    printable_require_item(require) for require in cast(List[Require], value)
```

Comment:
Using the `printable_require_item()` sounds a bit weird under `Core.export()` which should produce data not strings to be printed. I see the current implementation "works", but it seems this could be confusing. And could break, for example if `printable_require_item()` is updated to provide some more human-readable version of the beakerlib libraries. What do you think?

Reply:
This approach sounds good to me.

Reply:
I'm slightly inclided to do it here. But either works for me.

Reply:
Ok, I'll try to prepare an additional commit with this change.

Reply:
One preliminary patch though, https://github.com/teemtee/tmt/pull/1637 - `to_minimal_dict()` was incorrectly used in places where fmf id specification was expected, its predecessor was `to_spec()` but that one did include too many keys. I'd like to add `to_minimal_spec()` first, then rebase this patch and construct a `RequireSimple(str)` class for trivial requirements, implementing `SpecBasedContainer` methods we use when working with `RequireFmfId`. Then we can easily pretend all requirement types are ducks :)